tools/io-trace-explorer/io_trace.cc

```diff
@@ -0,0 +1,415 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <vector>
+
+#include "io_trace.h"
+
+/*
+ * io_trace::io_trace --
+ *     Initialize a new io_trace object.
+ */
+io_trace::io_trace(io_trace_collection &parent, const char *name) : _name(name), _parent(parent) {}
+
+/*
+ * io_trace::~io_trace --
+ *     Destroy a io_trace object.
+ */
+io_trace::~io_trace() {}
+
+/*
+ * io_trace_collection::io_trace_collection --
+ *     Initialize a new trace collection.
+ */
+io_trace_collection::io_trace_collection() {}
+
+/*
+ * io_trace_collection::~io_trace_collection --
+ *     Destroy a collection of traces.
+ */
+io_trace_collection::~io_trace_collection()
+{
+    for (auto &e : _traces) {
+        delete e.second;
+    }
+}
+
+/*
+ * io_trace_collection::add --
+ *     Add a data point.
+ */
+void
+io_trace_collection::add_data_point(
+  const std::string &device_or_file, const io_trace_kind kind, const io_trace_operation &item)
+{
+    std::string name = device_or_file;
+
+    /* Postprocess the data point. */
+
+    if (kind == io_trace_kind::WIRED_TIGER) {
+        const char *s = name.c_str();
+        if (strstr(s, "Log.00000") || strstr(s, "log.00000")) {
```

Comment:
We prefer to not mix C style string methods in with C++ code. 
- strstr -> name.find()
- strlen -> name.length()
- strcmp -> name.compare() or == if not requiring a substring match